lib/src/ui/screens/checkup/checkup_loaded_body.dart

```
@@ -19,18 +22,38 @@ class _CheckupLoadedBodyState extends State<CheckupLoadedBody> {
   CheckupStep currentStep = steps[0];
 
   void _saveCurrentLocation(CheckupStateInProgress checkupState) async {
-    final Geolocator geolocator = Geolocator();
-
+    bool showLocationFallback = true;
+    String postalCode;
     try {
-      final Position position = await geolocator.getCurrentPosition(
-        locationPermissionLevel: GeolocationPermission.locationWhenInUse,
-      );
-      final List<Placemark> places = await geolocator.placemarkFromCoordinates(
-        position.latitude,
-        position.longitude,
-      );
-      final String postalCode = places[0].postalCode;
-
+      final Geolocator geolocator = Geolocator();
+      final GeolocationStatus permission =
+          await geolocator.checkGeolocationPermissionStatus(
+              locationPermission: GeolocationPermission.locationWhenInUse);
+      if (permission == GeolocationStatus.granted ||
+          permission == GeolocationStatus.restricted) {
+        final Position position = await geolocator.getCurrentPosition(
+          locationPermissionLevel: GeolocationPermission.locationWhenInUse,
+        );
+        final List<Placemark> places =
+            await geolocator.placemarkFromCoordinates(
+          position.latitude,
+          position.longitude,
+        );
+        final String postalCode = places[0].postalCode;
+        showLocationFallback = false;
+      }
+    } catch (e) {
+      print(e.toString());
```

Do we need this print? @joshua-s do we have any logging or do we plan on doing any?

---

Currently, we do not have logging. We will need it eventually.

As far as this section is concerned. It would be good to fall back to the zip code prompt if geolocation throws an error.

---

As far as this logic is concerned, I'm not convinced that we will need to log this. What are your thoughts?

---

I don't think it's necessary as long as we only swallow the Geolocation exceptions.